baseapp/baseapp.go

```diff
@@ -609,6 +609,141 @@ func (app *BaseApp) cacheTxContext(ctx sdk.Context, txBytes []byte) (sdk.Context
 	return ctx.WithMultiStore(msCache), msCache
 }
 
+// runCheckTxConcurrently processes a transaction with either the checkTx or recheckTx modes, encoded transaction
```

Comment:
Do we have any tests for this function? Can we create a tech-debt ticket to track if you are working on higher priority item?

Reply:
I specifically added the [concurrent matches and cancels tests](https://github.com/dydxprotocol/v4/commit/5ad33f1e2cd54c378a23e05fc7056080ca43f91f) to cover most of the concurrent scenarios here, also [updated the simulator to use this function over runTx](https://github.com/dydxprotocol/cosmos-sdk/blob/6477dcf5693422fef693432e29bd979709aa995d/baseapp/test_helpers.go#L19), and we have some coverage from the CosmosSDKs tests themselves. I feel as though this would be sufficient but if you think there are certain scenarios you would want covered in addition to the above I would open up a ticket around this.

Reply:
I see, I agree that this feels sufficient for now. Thanks for clarifying!